extensions/arc/src/ui/dashboards/postgres/postgresComputeAndStoragePage.ts

```diff
@@ -0,0 +1,642 @@
+
+		this.memoryMaxBox = this.modelView.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
+			readOnly: false,
+			min: 0.25,
+			validationErrorMessage: loc.memoryMaxValidationErrorMessage,
+			inputType: 'number'
+		}).component();
+
+		this.memoryMinBox = this.modelView.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
+			readOnly: false,
+			min: 0.25,
+			validationErrorMessage: loc.memoryMinValidationErrorMessage,
+			inputType: 'number'
+		}).component();
+
+	}
+
+	private updateUserInputSection(): azdata.Component[] {
+		if (!this._postgresModel.configLastUpdated) {
+			return [this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+				value: loc.loading,
+				CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+			}).component()];
+		} else {
+
+			this.editWorkerNodeCount();
+			this.editVCores(loc.vCoresMin);
+			this.editVCores(loc.vCoresMax);
+			this.editMemory(loc.memoryMin);
+			this.editMemory(loc.memoryMax);
+
+			return [
+				this.workerNodesSectionContainer(),
+				this.vCoreMemorySection(),
+				this.configurationSectionContainer(loc.vCoresMin, this.vCoresMinBox!, '40px'),
+				this.configurationSectionContainer(loc.vCoresMax, this.vCoresMaxBox!, '40px'),
+				this.configurationSectionContainer(loc.memoryMin, this.memoryMinBox!, '40px'),
+				this.configurationSectionContainer(loc.memoryMax, this.memoryMaxBox!, '20px')
+
+			];
+
+		}
+
+
+	}
+
+	private workerNodesSectionContainer(): azdata.FlexContainer {
+		const inputFlex = { flex: '0 1 150px' };
+		const keyFlex = { flex: `0 1 250px` };
+
+		const flexContainer = this.modelView.modelBuilder.flexContainer().withLayout({
+			flexWrap: 'wrap',
+			alignItems: 'center'
+		}).component();
+
+		const keyComponent = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: loc.workerNodeCount,
+			CSSStyles: { ...cssStyles.text, 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const keyContainer = this.modelView.modelBuilder.flexContainer().withLayout({ alignItems: 'center' }).component();
+		keyContainer.addItem(keyComponent, { CSSStyles: { 'margin-right': '0px', 'margin-bottom': '15px' } });
+
+		const information = this.modelView.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			iconPath: IconPathHelper.information,
+			title: loc.workerNodesInformation,
+			width: '12px',
+			height: '12px',
+			enabled: false
+		}).component();
+
+		keyContainer.addItem(information, { CSSStyles: { 'margin-left': '5px', 'margin-bottom': '15px' } });
+		flexContainer.addItem(keyContainer, keyFlex);
+
+		const inputContainer = this.modelView.modelBuilder.flexContainer().withLayout({ alignItems: 'center' }).component();
+		inputContainer.addItem(this.workerBox!, { CSSStyles: { 'margin-bottom': '15px', 'min-width': '50px', 'max-width': '225px' } });
+
+		flexContainer!.addItem(inputContainer, inputFlex);
```

Comment:
Make sure all the places using `!.` actually need it.  Looks like it's used in some places where the compiler should be smart enough to know the variable is defined.  Like in this case where flexContainer is defined above.